test/controllers/response_sets_controller_test.rb

```diff
@@ -146,4 +128,25 @@ class ResponseSetsControllerTest < ActionDispatch::IntegrationTest
     assert response_json['surveillance_systems'].include? 'National Insignificant Digits System'
     assert response_json['surveillance_programs'].include? 'Generic Surveillance Program'
   end
+
+  test 'publishers should see response_sets from other authors' do
+    sign_out @current_user
+    @current_publisher = users(:publisher)
+    sign_in @current_publisher
+    get response_set_url(response_sets(:two), format: :json)
+    assert_response :success
+  end
+
+  test 'publishers should be able to publish response_sets' do
+    sign_out @current_user
+    @current_publisher = users(:publisher)
+    sign_in @current_publisher
+    put publish_response_set_path(response_sets(:two), format: :json)
+    assert_response :success
```

Comment:
In the question test you check if the object status is PUBLISHED, why not do that for every controller? Also, maybe should also check that the publisher attribute is properly set to the publisher doing the publishing, but I could be convinced this is handled by other tests.